jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -465,18 +470,20 @@ def _plot_mask(self):
 
         # Loop through all active data in the viewer
         for index, layer_state in enumerate(self.state.layers):
-            comps = [str(component) for component in layer_state.layer.components]
+            lyr = layer_state.layer
+            comps = [str(component) for component in lyr.components]
 
             # Skip subsets
-            if hasattr(layer_state.layer, "subset_state"):
+            if hasattr(lyr, "subset_state"):
                 continue
 
             # Ignore data that does not have a mask component
             if "mask" in comps:
-                mask = np.array(layer_state.layer['mask'].data)
+                mask = np.array(lyr['mask'].data)
 
-                data_x = layer_state.layer.data.get_object().spectral_axis
-                data_y = layer_state.layer.data.get_object().flux.value
+                data_obj = lyr.data.get_object()
+                data_x = data_obj.spectral_axis.value
```

Comment:
Pretty sure you want the underlying numpy array, not the Quantity right? I think the missing `.value` here might be a bug so I added it but please confirm. Thanks!